package.json

```diff
@@ -32,23 +32,15 @@
     "shower-ribbon": "^3.1.2"
   },
   "devDependencies": {
-    "browser-sync": "^2.24.5",
-    "del": "^3.0.0",
-    "gh-pages": "^1.2.0",
-    "gulp": "^4.0.0",
-    "gulp-rename": "^1.2.2",
-    "gulp-replace": "^1.0.0",
-    "gulp-zip": "^4.0.0",
-    "merge-stream": "^1.0.0",
     "path-exists-cli": "^1.0.0",
-    "shower-cli": "^0.1.4"
+    "shower-cli": "^0.2.2"
   },
   "scripts": {
     "pdf": "shower pdf",
-    "start": "gulp serve",
-    "prepare": "gulp prepare",
-    "archive": "gulp archive",
-    "publish": "gulp publish",
+    "start": "shower serve",
+    "prepare": "shower prepare --files '**' --files '!{docs,node_modules,prepared}{,/**}' --files '!*.{md,toml,json}'",
+    "archive": "shower archive --files '**' --files '!{docs,node_modules,prepared}{,/**}' --files '!*.{md,toml,json}'",
+    "publish": "shower publish --files '**' --files '!{docs,node_modules,prepared}{,/**}' --files '!*.{md,toml,json}'",
```

Comment:
On existing gh-pages branch everything works fine, but once it’s deleted I get this:

```
~/Documents/Shower/shower on migrate-to-cli
$ npm run publish

> shower@2.6.2 publish /Users/pepelsbey/Documents/Shower/shower
> shower publish --files '**' --files '!{docs,node_modules,prepared}{,/**}' --files '!*.{md,toml,json}'

 ✖ The project is being published
   → fatal: Couldn't find remote ref refs/heads/gh-pages
shower publish

- Publish the presentation to gh-pages

Options:
  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]
  --files, -f  List of files that will get the build                     [array]

{ ProcessError: fatal: Couldn't find remote ref refs/heads/gh-pages

    at ChildProcess.<anonymous> (/Users/pepelsbey/Documents/Shower/shower/node_modules/gh-pages/lib/git.js:42:16)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:961:16)
    at Socket.stream.socket.on (internal/child_process.js:380:11)
    at Socket.emit (events.js:182:13)
    at Pipe._handle.close (net.js:599:12)
  code: 128,
  message: 'fatal: Couldn\'t find remote ref refs/heads/gh-pages\n',
  name: 'ProcessError',
  context: {} }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! shower@2.6.2 publish: `shower publish --files '**' --files '!{docs,node_modules,prepared}{,/**}' --files '!*.{md,toml,json}'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shower@2.6.2 publish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pepelsbey/.npm/_logs/2018-08-31T06_51_30_726Z-debug.log
```

Reply:
So it needs to check if it exists and if not create a branch, not fail with an error.

Reply:
You can help me if you tell me how to reproduce the problem with publish — the current version creates a branch if:
- no local or remote;
- there is a local no remote;
- there is a local with a link to a non-existent remote;
- there is no local remote.

Reply:
Right, the non-existent branch problem is still there. Steps to reproduce:

1. Delete `origin/gh-pages` branch. The remote one, from your GitHub.
2. Run `npm run publish` and you’ll get:

```
$ npm run publish

> shower@2.6.8 publish /Users/pepelsbey/Documents/Shower/shower
> shower publish --files '**' --files '!{docs,node_modules,prepared}{,/**}' --files '!*.{md,toml,json}'

shower publish

Publish the presentation to gh-pages

Options:
  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]
  --files, -f  List of files that will get the build                     [array]

{ ProcessError: fatal: Couldn't find remote ref refs/heads/gh-pages

    at ChildProcess.<anonymous> (/Users/pepelsbey/Documents/Shower/shower/node_modules/shower-cli/node_modules/gh-pages/lib/git.js:42:16)
    at ChildProcess.emit (events.js:197:13)
    at maybeClose (internal/child_process.js:978:16)
    at Socket.stream.socket.on (internal/child_process.js:395:11)
    at Socket.emit (events.js:197:13)
    at Pipe._handle.close (net.js:611:12)
  code: 128,
  message: "fatal: Couldn't find remote ref refs/heads/gh-pages\n",
  name: 'ProcessError' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! shower@2.6.8 publish: `shower publish --files '**' --files '!{docs,node_modules,prepared}{,/**}' --files '!*.{md,toml,json}'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the shower@2.6.8 publish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pepelsbey/.npm/_logs/2019-02-10T13_29_31_217Z-debug.log
```

Reply:
It needs to create remote `gh-pages` branch in case it doesn’t exist. Otherwise it fails to publish to it.

Reply:
Hmmm... Strange problem, I still can't reproduce. I made a test presentation to make sure there was a problem. Here's what I do: ![Описание картинки с котиком](https://lh3.googleusercontent.com/kbX1HVXzfyMg9Mrbm04z7qpeYG63ME0i9Ml73gtItIoKjxM01ZpPyU4DPCDQ7U8VSmbdSKun-5lFl0C0QMCHTbusTFTiHp1KUz8D3-AaAsmDnFCXieD2pZc6e_67MLQcGorVH0MUzX1O-GGox48g1CEzkMsBh0d5d607bCGP9kSla44tiGGPoVmiFWfW2vkiIgc_q5_fRDg9K9ZMXYNBZO6F-fu2qvIPnCeZ8ilyafvFWBjPeCYexO1_tAf-ewyRcfcfF2tFQ9JPA7MDKFnW7ho6a5WFqJPMGaJgt0IZh3hWh-ADZQcZk0_A-GyJIcJ7pvYs_ogWCd98XTuN200dXxXsOH_RzHOkO0FwWht1oyFlpJgRn-QQXJsCIm8ey_uRC_2ZzhPLkzBR8a0RElD1UM5DtF2fvaqSAnelwPGCtb8QBclaGs4wKzcSVtuzU9ZT-TNq1sNxlKshHIpXvhiwGrVjXQ_8BrXmr5Rmpeh5Im4uU1475fXY4KqDLZwpVbyHvL9Xc5aVQq71_hx7WNkrkM3Kv512vboI783Xliy1qrGOt8cm-a-yi2UYrq0m58kegKhm1K2Vr7xVWylhSFDtmKTqwmxdUG-CO8EHLYgvQHqW0KHn0cjcfEoLXLL0bNPDIvno2iVklZONb0SaxTO97IjEGZODHzmCCHWelneAnxgCsMoWzJnT3LzVL4I_zUPHIcsE-uhwPnG9q55lqOJuzt5m=w1366-h580-no)

I found a description of this problem in the "gh-pages" reports": https://github.com/tschaub/gh-pages/issues/17. Maybe this will help you?